components/table/README.md

```diff
@@ -299,3 +299,87 @@ If your table supports row selection, apply the `selected` attribute to `<tr>` r
   <td>this row is selected</td>
 </tr>
 ```
+
+## Table Header [d2l-table-header]
+
+The `d2l-table-header` component can be placed in the `d2l-table-wrapper`'s `header` slot to provide a selection summary, a slot for `d2l-selection-action`s, and overflow-group behaviour.
+
+<!-- docs: demo live name:d2l-table-header autoSize:false size:medium -->
+```html
+<script type="module">
+  import '@brightspace-ui/core/components/selection/selection-action.js';
+  import '@brightspace-ui/core/components/selection/selection-input.js';
+  import '@brightspace-ui/core/components/selection/selection-select-all.js';
+  import '@brightspace-ui/core/components/table/table-header.js';
+  import { html, LitElement } from 'lit';
+  import { tableStyles } from '@brightspace-ui/core/components/table/table-wrapper.js';
+
+  class MyTableWithHeaderElem extends LitElement {
+
+    static get properties() {
+      return {
+        _checked: { state: true }
+      }
+    }
+
+    static get styles() {
+      return tableStyles;
+    }
+
+    constructor() {
+      super();
+      this._checked = [ true, false ];
+    }
+
+    render() {
+      return html`
+        <d2l-table-wrapper>
+          <d2l-table-header slot="header" no-sticky>
+            <d2l-selection-action icon="tier1:delete" text="Delete" requires-selection></d2l-selection-action>
+            <d2l-selection-action icon="tier1:gear" text="Settings"></d2l-selection-action>
+          </d2l-table-header>
+          <table class="d2l-table">
+            <thead>
+              <tr>
+                <th><d2l-selection-select-all></d2l-selection-select-all></th>
+                <th>Column B</th>
+              </tr>
+            </thead>
+            <tbody>
+              <tr selected>
+                <td>
+                  <d2l-selection-input selected key="a" label="a" ?checked="${this._checked[0]}" @d2l-selection-change="${() => this._selectRow(0)}"></d2l-selection-input>
```

Comment:
I think in practice the linter/analyzer might complain about calling functions in handlers defined in templates. I think it's a perf thing to avoid the recreation of functions when the templates re-render. To avoid it, we'd normally use the same handler for all the `d2l-selection-input`, and use `e.target.key` to do the necessary look-up.